File: benchmarks/bench_neuron.py
import time, sys
def relu(x):
    return x if x > 0 else 0

def layer(inputs, neurons):
    total = 0
    for n in range(neurons):
        acc = 0
        for i in range(inputs):
            w = (n * inputs + i + 1) % 11
            x = (i - inputs // 2) * w
            acc += relu(x)
        total += acc
    return total

t0 = time.perf_counter()
result = 0
for _ in range(500):
    result = layer(128, 256)
ms = (time.perf_counter() - t0) * 1000
print(result)
print(f'COMPUTE_MS={ms:.3f}', file=sys.stderr)

File: benchmarks/bench_neuron.js
const { performance } = require('perf_hooks');
function relu(x) { return x > 0 ? x : 0; }
function layer(inputs, neurons) {
    let total = 0;
    for (let n = 0; n < neurons; n++) {
        let acc = 0;
        for (let i = 0; i < inputs; i++) {
            const w = (n * inputs + i + 1) % 11;
            const x = (i - (inputs >> 1)) * w;
            acc += relu(x);
        }
        total += acc;
    }
    return total;
}
const t0 = performance.now();
let result = 0;
for (let rep = 0; rep < 2000; rep++) result = layer(128, 256);
const ms = performance.now() - t0;
console.log(result);
process.stderr.write(`COMPUTE_MS=${ms.toFixed(3)}\n`);
